WordPress/gutenberg
# Font Library: Improve usability of font variant selection

## What?

This updates the font variant markup so that the checkbox controls are wrapped inside a label element. This means that any part of the font variant panel can be clicked to interact with the checkbox, rather than only the checkbox itself.

## Why?

This improves the usability of these checkboxes, as you can click on a much larger area for the toggle interactions. It also should improve their accessibility, as previously these checkboxes did not include any labels. Now when a checkbox is read out by a screenreader, it will read out the font variant name.

## How?

There are two main updates: updating the markup of the font variant selection for both the library and collection versions; and updating the `CheckboxControl` component to include the option for preventing a label from being rendered, in order to allow for a custom label to be used outside the component.

## Testing Instructions




1. Ensure you're using a theme that includes at least one bundled font family (or upload a font family as part of this test)
2. Open the Font Library modal
3. Open a font family
4. Toggle a font variant of the selected font family by clicking anywhere in the font variant panel, rather than just the checkbox

### Testing Instructions for Keyboard

1. Ensure you're using a theme that includes at least one bundled font family (or upload a font family as part of this test)
2. Open the Font Library modal
3. Open a font family
4. Toggle a font variant of the selected font family by tabbing to the font variant panel and toggling the checkbox. If using a screenreader, the font variant name should be called out.

## Screenshots or screencast 

| Before | After |
| ------ | ------ |
| <video src="https://github.com/WordPress/gutenberg/assets/1645628/089895ec-f222-4883-8477-534fe560678d" /> | <video src="https://github.com/WordPress/gutenberg/assets/1645628/da7abdf4-20ae-449c-8e7a-321adc348fa2" /> |
